backend/extensions/Squidex.Extensions/Actions/SignalR/SignalRActionHandler.cs

```diff
@@ -0,0 +1,153 @@
+﻿// ==========================================================================
+//  Squidex Headless CMS
+// ==========================================================================
+//  Copyright (c) Squidex UG (haftungsbeschraenkt)
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Azure.SignalR.Management;
+using Squidex.Domain.Apps.Core.HandleRules;
+using Squidex.Domain.Apps.Core.Rules.EnrichedEvents;
+
+namespace Squidex.Extensions.Actions.SignalR
+{
+    public sealed class SignalRActionHandler : RuleActionHandler<SignalRAction, SignalRJob>
+    {
+        private readonly ClientPool<(string ConnectionString, string HubName), IServiceManager> clients;
+
+        public SignalRActionHandler(RuleEventFormatter formatter)
+            : base(formatter)
+        {
+            clients = new ClientPool<(string ConnectionString, string HubName), IServiceManager>(key =>
+            {
+                var serviceManager = new ServiceManagerBuilder()
+                    .WithOptions(option =>
+                    {
+                        option.ConnectionString = key.ConnectionString;
+                        option.ServiceTransportType = ServiceTransportType.Transient;
+                    })
+                    .Build();
+                return serviceManager;
+            });
+        }
+
+        protected override async Task<(string Description, SignalRJob Data)> CreateJobAsync(EnrichedEvent @event, SignalRAction action)
+        {
+            var hubName = await FormatAsync(action.HubName, @event);
+
+            string requestBody;
+
+            if (!string.IsNullOrWhiteSpace(action.Payload))
+            {
+                requestBody = await FormatAsync(action.Payload, @event);
+            }
+            else
+            {
+                requestBody = ToEnvelopeJson(@event);
+            }
+
+            string[] users = new string[0];
```

Comment:
Don't make it so complicated ;)

```
var users = await FormatAsync(action.User, @event);
var userArray = users.Split('\n');
```

And then in the execute

```
if (job.Users.Count == 1)
{
    await signalRContext.Clients.User(job.User).SendAsync(methodeName, job.Payload);
}
else
{
    // But what happens if you pass in only one user? Do you really need the if?
    await signalRContext.Clients.Users(job.Users).SendAsync(methodeName, job.Payload);
}
```

Reply:
In fact I wanted to shield the transition from the object to the Client methods.
User can only take a string as parameter, Users can only take an array of strings. Afterwards, if the Array for Users contains only one item, it's not a problem